File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>
#include <cmath>
#include <algorithm>

using namespace std;

vector<int> solution(string s) {
    vector<int> answer;
    int count = 0, zeroCount = 0;
    
    while(s != "1")
    {
        string temp = "";
        count++;
        
        for(int i = 0; i < s.length(); i++)
        {
            if(s[i] == '1')
                temp += "1";
            else
                zeroCount++;
        }
        
        int len = temp.length();
        s.clear();
        
        while(len > 0)
        {
            s += to_string(len % 2);
            len /= 2;
        }
    }
    
    answer.push_back(count);
    answer.push_back(zeroCount);
    return answer;
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        int[] answer = new int[2];
        int count = 0, zeroCount = 0;
        
        while(!s.equals("1"))
        {
            int temp = 0;
            count++;
            
            for(int i = 0; i < s.length(); i++)
            {
                if(s.charAt(i) == '1')
                    temp++;
                else
                    zeroCount++;
            }
            
            s = Integer.toBinaryString(temp);
        }
        
        answer[0] = count;
        answer[1] = zeroCount;
        
        return answer;
    }
}